app/src/main/java/com/example/taskyy/data/repositories/AgendaRepositoryImpl.kt

```diff
@@ -74,6 +97,39 @@ class AgendaRepositoryImpl @Inject constructor(
         }
     }
 
+    override suspend fun updateReminderToApi(reminder: Reminder): Result<Reminder, DataError.Network> {
+        val reminderDTO = reminder.toReminderDto()
+        return try {
+            retrofit.updateReminder(reminderDTO)
+            Result.Success(reminder)
+        } catch (e: HttpException) {
+            when (e.code()) {
+                408 -> Result.Error(DataError.Network.REQUEST_TIMEOUT)
+                429 -> Result.Error(DataError.Network.TOO_MANY_REQUESTS)
+                413 -> Result.Error(DataError.Network.PAYLOAD_TOO_LARGE)
+                500 -> Result.Error(DataError.Network.SERVER_ERROR)
+                400 -> Result.Error(DataError.Network.SERIALIZATION)
+                else -> Result.Error(DataError.Network.UNKNOWN)
+            }
```

Comment:
This response code handling could perfectly be outsourced to a helper function, since it'll be the same for every single HTTP call